lib/data/repositories/utils/executor.dart

```diff
@@ -1,25 +1,30 @@
 import 'package:chopper/chopper.dart';
-import 'package:coffeecard/errors/request_error.dart';
+import 'package:coffeecard/data/repositories/utils/request_types.dart';
 import 'package:coffeecard/utils/either.dart';
-import 'package:coffeecard/utils/extensions.dart';
 import 'package:logger/logger.dart';
 
+/// Helper class that executes network requests and handles errors
 class Executor {
-  Future<Either<RequestError, T>> executeNetworkRequestSafely<T>(
-    Future<Response<T>> Function() f,
-    Logger logger,
-  ) async {
-    final Response<T> response;
+  const Executor(this.logger);
+  final Logger logger;
+
+  /// Executes a network request and handles errors, including IO errors.
+  /// Returns either a [RequestError] or the transformed success type.
+  Future<Either<RequestError, Ret>> execute<Dto, Ret>(
+    Future<Response<Dto>> Function() request, {
+    required Ret Function(Dto dto) transformer,
```

Comment:
Why are we giving the parameters in two different ways?

Reply:
both are now positional